drivers/platform/pico/pico_spi.c

```diff
@@ -52,62 +52,36 @@
 /************************ Functions Definitions *******************************/
 /******************************************************************************/
 
-/**
- * @brief Initialize the SPI communication peripheral.
- * @param desc  - The SPI descriptor.
- * @param param - The structure that contains the SPI parameters.
- * @return 0 in case of success, error code otherwise.
- */
-int32_t pico_spi_init(struct no_os_spi_desc **desc,
-		      const struct no_os_spi_init_param *param)
+static int pico_spi_config(struct no_os_spi_desc *desc)
 {
-	int32_t ret;
-	struct no_os_spi_desc *descriptor;
 	struct pico_spi_desc *pico_spi;
-	struct pico_spi_init_param *pico_spi_ip;
+	pico_spi = desc->extra;
```

Comment:
Maybe assign it directly on the previous line, or below the variable declaration block.